deploy/kubernetes/elastic-agent-kustomize/ksm-hints/elastic-agent-standalone/kustomization.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../../default/elastic-agent-standalone
+
+patches:
+- patch: |-
+    apiVersion: apps/v1
+    kind: DaemonSet
+    metadata:
+      name: elastic-agent-standalone
+      namespace: kube-system
+      labels:
+        app: elastic-agent-standalone
+    spec:
+      template:
+        spec:
+          initContainers:
+            - name: k8s-templates-downloader
+              image: docker.elastic.co/beats/elastic-agent:9.0.0
+              command: ['bash']
+              args:
+                - -c
+                - >-
+                  mkdir -p /usr/share/elastic-agent/state/inputs.d &&
+                  curl -sL https://github.com/elastic/elastic-agent/archive/9.0.tar.gz | tar xz -C /usr/share/elastic-agent/state/inputs.d --strip=5 "elastic-agent-9.0/deploy/kubernetes/elastic-agent-standalone/templates.d"
+              securityContext:
+                runAsUser: 0
```

Comment:
I remember back in the day that the hints approach was using an emptydir volume that the init container would download and move the templates there and then the agent container would mount it on the appropriate path. This approach allowed the init container to run as non root. Any reason this approach isn't still relevant?

Reply:
It is the same approach, only that now we use the already mounted `/usr/share/elastic-agent/state/` path now and we dont create another folder
This changed since this [PR](https://github.com/elastic/elastic-agent/pull/4625)

Reply:
what I have in mind is the following; First running as root with all capabilities an init container seems kinda excessive, with emptydir this wasn't necessary. Second, if the pod gets restarted, now that you write the templates directly in the state of the agent which a host mountpath that is already populated by the previous pod execution, it means that the init container is prone to face permission errors, as elastic-agent container could have changed the permissions of its state dir (agent does that after [this PR](https://github.com/elastic/elastic-agent/pull/4925)).

Reply:
Yes it is a scenario. Of course this approach with the init container was a general solution. We document [here](https://www.elastic.co/guide/en/fleet/current/hints-annotations-autodiscovery.html#_configure_hints_autodiscovery) that this can be done manually too. 

I guess what you propose above to already copy them in the image will solve our problems

Reply:
I think I got what we can do here. 

In the initContainer we can set the 
```
volumeMounts:
        - mountPath: /etc/elastic-agent/inputs.d
          name: external-inputs
```
where
```
- name: external-inputs
        emptyDir: {}
```
and untar the templates in there. It will be started with default permissions (not root)

The main container will just have
```
- name: external-inputs
          mountPath: /usr/share/elastic-agent/state/inputs.d
```

I think that covers all scenarios. WDYT?

Reply:
@MichaelKatsoulis I am still wondering why this needs to write in a hostpath and thus require "root"; AFAICT the init container everytime that the pod restarts will re-download the templates and because of that this path can be easily considered ephemeral. I think what you are taking for granted here is that the user will allow all the capabilities to the root user but this is not always the case, especially for an init container. So still an emptydir that gets mounted at `/usr/share/elastic-agent/state/inputs.d` seems more ideal to me and it doesn't require your init container to run as root. Also bear in mind that there is an active push to run elastic-agent as rootless and not rootful. Now my in-the-future suggestion, is to add these templates in the agent image (or create a variant of it at least) as their size is really small. Feel free to ask more question, more than happy to reply 🙂

Reply:
> I think that covers all scenarios. WDYT?
this is what I had in mind @MichaelKatsoulis indeed! If you could could test that everything works fine with the emptyDir approach I think it would be better 🙂 (please for both rootful and rootless agent invocations)

Reply:
Running elastic-agent without root permissions is trickier as agent tries to create data folder in a host path. Maybe it needs some specific psp. Do you know if there is such an example? I can just set the state volume to an empty dir instead. 
It works in both scenarios. Rootful and rootless agent. I will update the configuration in this PR